### DISCUSSION TITLE
Spring 2022 Hackathon

### DISCUSSION THREAD

**Discussion by BekahHW** [Virtual-Coffee/VC-Community-Docs#190] [Ideas]
Let's do some brainstorming for what a VC Hackathon would look like!

**Comment by marktnoonan** [comment]
So I kind of have a baseline of projects we can look to as "non-profit/social good/accessibility" projects where I could see packaging up some tasks/needs and making those available for participants/teams to grab and work on. If we even have formal teams. But I guess that's a whole other discussion about how this would run, how people would be supported etc. 

For reference here's my favorite ever non-profit focused hackathon: https://medium.com/code-for-atlanta/atlanta-greenspace-hackathon-prototypes-ead39f8bdcf2

Initial Project ideas that I _know_ we have the ability/connections to support:

- CanopyATL.org - nonprofit org - they have lots of ideas, the low hanging fruit is website accessibility for their wordpress site, I just talked with them and they are totally up for this and excited to see if they can form any of their other needs into suitable projecs
- Peoplemakingprogress.org - could use a fresh coat of paint, new donation flow, and actually the schedule email system I wrote 5 years ago for them could really be rewritten with a better UI. So lots of options there.
- AutismActivities.ie is a site I own, we have a v2 planned but I would also like to generalize this Storyblok CMS + nuxt + netlify situation into something that can be stood up easily by anybody for similar kinds of sites
- FreeLiveTranscript.com could use a ton of work, lots of potenial.

The goals of the hackathon the way I see it would be:

- make some useful contributions to good projects around A11y/Access/Nonprofit work that we care about
- get some awards/acknowledgement/experience onto resumes of folks in VC
- make all the participants feel like the rockstars they are, by correctly mentoring/supporting them during the event in the way that only VC can

**Reply by tkshill** [reply]
I love these resources @marktnoonan 

I love the idea of themes centered around volunteerism and social good.
I don't have strong opinions of teams yet. 

I'd like for another goal to be a "healthy hackathon"
- One month for the time, but we scope projects that should take a few hours a week max
- A focus on good dev principals. Testing. Version Control. Documentation. Descriptive PRs.

**Reply by marktnoonan** [reply]
> One month for the time, but we scope projects that should take a few hours a week max

Totally agree with this. Or at least, a range of task sizes so that somebody who only has a few hours in the month can still complete some small, meaningful unit of work.

> A focus on good dev principals. Testing. Version Control. Documentation. Descriptive PRs.

I think we'd have to be delicate with this. The feedback cycles involved could complicate things and make it a lil more intimidating, and may place a lot of burden on whoever is involved in a more mentorship role. Buuuut on the other hand a lot of work will probably not be throwaway prototypes like a lot of hackathons, so we may be able to think about like, having folks do POCs as part of the event and then opt in to feedback/guidance for how to get those "production ready". I looooove the ridiculous, "let's put this helicopter together with duct tape and fly it" experience of hackathons and I would be keen not to lose that. It's super fun for people to stretch waaaay beyond their skill level, but only a subset of those people will be ready to learn/do a bunch of meta stuff and that can be ok.

All of that said, a few basic ground rules about git process etc sounds neat, and maybe some incentive/recognition for code quality, test coverage, documentation, etc could make that all more fun for people.

**Reply by danieljanderson** [reply]
I like the idea of volunteerism too.  What do you guys think about it being open-ended because different communities have different issues? 

What do you guys feel about this:  Instead of giving people ideas they pitch the ideas.  The participants think of issues they have in their community and come back to the group and say "I have seen this y issue in my community.  If I could build x then it would solve y."  The people who feel passionate about solving for y go and help build x.

**Reply by calendee** [reply]
I like @danieljanderson's idea. Basically, we could "adopt" a project(s) after we close the pitching.

**Reply by marktnoonan** [reply]
I like that as well. I do think it's best to start small though, and a fully open even could get out of hand. I'd suggest having a predefined list of "problems to solve" which could absolutely include "I have seen this y issue in my community. If I could build x then it would solve y" type projects, but we'd just need to document them in advance and make sure that things are actually ready to work on. So maybe we'd have some sort of CFP period, explaining what is needed if you want to pitch and help bring to life a project for the month of the hackathon?

I'd also like to make sure that, if we do have nonprofits involved, they get some sort of priority since if we spend time shaping up the tasks and requirements with a certain nonprofit, and involve them with judging solutions etc, we _have_ to deliver something in order to make participating a good experience, not a waste of their time. So a wide breadth of projects could threaten that. It all depends on what the ratio of projects to participants turns out to be though, we have a lot of people. But as organizers, how many projects/teams/ideas can we support during the event? It would be good to have some limits.

**Reply by danieljanderson** [reply]
@marktnoonan what we could do is have people pitch their ideas a month or so in advance.  They would need to have some facts/ statistics to justify why this is an issue and then how their idea would fix the issue and how they would implement it.  They then vote and the top x amount of projects get worked on.  And if someone really wants their project picked than they can work on it.
